src/cpp/src/sampler.cpp

```diff
@@ -956,7 +956,7 @@ int64_t Sampler::GroupBeamSearcher::Group::finish(Beam beam, const ov::genai::Ge
     return preeempted_sequence_id;
 }
 
-void Sampler::GroupBeamSearcher::Group::is_done(const ov::genai::GenerationConfig& sampling_params) {
+void Sampler::GroupBeamSearcher::Group::is_done(const ov::genai::GenerationConfig& sampling_params, size_t prompt_length) {
```

Comment:
- `Sampler::GroupBeamSearcher::Group` holds `Beams`
- each `Beam` holds its `Sequence`
- each `Sequence` has a pointer to containing `SequenceGroup`
- `SequenceGroup` will have `get_max_new_tokens` method

Looks like we can drop all arguments from this method.

Reply:
BTW, maybe we can perform such changes as a separate (follow-up) PR?
in other places we also have max_new_tokens which should be replaced new your solution.